### PR TITLE
Feat(engine): Use new math API host functions

### PR DESCRIPTION
**Pull request by birchmd**
Updates our contract to use the new host functions merged in https://github.com/near/nearcore/pull/4380

Note: these host functions have not yet been released to mainnet, but they are on testnet. The host functions are in protocol version 46; you can see [here](https://explorer.testnet.near.org/stats) that 46 is the version of testnet, while [mainnet](https://explorer.near.org/stats) is still on 45.

Using the host functions (specifically `ecrecover`) makes simple ETH transfer on Aurora less than 1/5th the cost in terms of NEAR gas.

Requires #193 before merging.

### CODE REVIEW

File: src/sdk.rs [hunk 1]
```diff
@@ -372,6 +382,47 @@ pub fn keccak(input: &[u8]) -> H256 {
...
+    }
+}
+
+/// Recover address from message hash and signature.
+pub fn ecrecover(hash: H256, signature: &[u8]) -> Result<Address, ECRecoverErr> {
+    unsafe {
+        let hash_ptr = hash.as_ptr() as u64;
+        let sig_ptr = signature.as_ptr() as u64;
+        let recover_register_id = 1;
+        let keccak_register_id = 2;
```
Comment (joshuajbouw) [inline src/sdk.rs:402]: Should be consts.

File: src/sdk.rs [hunk 2]
```diff
@@ -372,6 +382,47 @@ pub fn keccak(input: &[u8]) -> H256 {
...
+        let result = exports::ecrecover(
+            32,
+            hash_ptr,
+            64,
+            sig_ptr,
+            signature[64] as u64,
+            0,
+            recover_register_id,
+        );
```
Comment (joshuajbouw) [inline src/sdk.rs:411]: Any value that is concrete, we should be making consts for.

**Review by joshuajbouw — CHANGES_REQUESTED** [review]
These should use the testnet feature so that we don't accidentally use it prematurely.

File: src/sdk.rs (hunk 1 above)
Comment (birchmd) [reply src/sdk.rs]: Done in https://github.com/aurora-is-near/aurora-engine/pull/190/commits/ede8a4df16419d61b2d50f301b2eae188324c1d3

File: src/sdk.rs (hunk 2 above)
Comment (birchmd) [reply src/sdk.rs]: Done in https://github.com/aurora-is-near/aurora-engine/pull/190/commits/ede8a4df16419d61b2d50f301b2eae188324c1d3

**Comment by birchmd** [comment]
> testnet feature

Done in https://github.com/aurora-is-near/aurora-engine/pull/190/commits/5c5544bd00051b8e53795d0f08137ee1742c2992

**Comment by joshuajbouw** [comment]
Requires https://github.com/aurora-is-near/aurora-engine/pull/193 before merging.

**Review by joshuajbouw — APPROVED** [review]

**Pull request merged**